### DISCUSSION TITLE
Performance of Docker

### DISCUSSION THREAD

**Discussion by christopherpeers** [cyberbotics/wrestling#53] [General]
Hello,

Recently I noticed some performance changes on the docker which started causing slow downs and such when running on the leaderboard. This of course then affects motions that are developed on local simulations making development tedious as the motion will behave differently when committed to leaderboard.

Would it be possible to clarify or fix this?

Thanks

**Comment by omichel** [comment]
We didn't change anything on the self-hosted runner that may affect the performance of the simulation running in the docker. However, since it is a standard Linux machine, it may be possible that some task in the background, like a security update, may randomly affect the performance of the simulation. This is unfortunately difficult to avoid.

It may also be possible that some robot controllers are becoming increasingly computer intensive and we currently do not limit the amount of CPU cores allocated to each controller process (although we planned to limit it to 3 cores and 6 GB or RAM as documented in the [rules](https://github.com/cyberbotics/wrestling#runner-machine-configuration)).

So, we will implement this limitation to ensure that a controller program cannot slow down the simulation and cannot slow down its opponent.

**Comment by omichel** [comment]
The limitation was just implemented [here](https://github.com/cyberbotics/competition-record-action/blob/main/metascript/animation.py#L194-L197) and [here](https://github.com/cyberbotics/wrestling/blob/main/webots.yml#L9-L10). This should prevent participant controllers from eating up all CPU or memory, slowing down the simulation and/or their opponent.
